tensorflow/python/ops/array_ops.py

```diff
@@ -2454,6 +2455,88 @@ def _all_dimensions(x):
   return range(0, rank(x))
 
 
+@tf_export("multi_one_hot")
+def multi_one_hot(indices, depth_list, on_values_list=None,
+                  off_values_list=None, name=None):
+  """Creates one-hot-encodings for multiple features (columns in a matrix)
+  and concatenates the resulting encodings
+
+  If `indices` is a matrix (batch) with shape `[batch, features]`, the output
+  shape will be `[batch, sum(depth_list)]`:
+
+  Example
+  =========
+
+  Suppose that
+
+  ```python
+    indices =
+    [1, 2]
+    [0, 1]
+    [1, 0]
+
+    depth_list = [2, 3]
+    on_values_list = [5.0, 10.0]
+    off_value_list = [0.0, 0.0]
+  ```
+
+  Then output is `[3 x (2+3)]` which is `[3 x 5]`:
+
+  ```python
+    output =
+    [0.0 5.0]                     [ 0.0  0.0 10.0]
+    [5.0 0.0]  concatenated with  [ 0.0 10.0  0.0]
+    [5.0 0.0]                     [10.0  0.0  0.0]
+  ```
+
+  Args:
+    indices: A `Tensor` with mutiple columns.
+    depth_list: A list containing the depth for each column.
```

Comment:
So just to make sure I understand: in the case where `depth_list` is length-1, this reduces to normal one-hot, right?

Reply:
If the length of `depth_list` is 1, we face a regular one-hot-encoding, if this is what you mean.